spec/app/ics-101-interchain-swap/README.md

```diff
@@ -227,54 +245,71 @@ class InterchainMarketMaker {
         const weight = asset.weight / 100
         const issueAmount = supply * (math.pow(1+amount/asset.balance, weight) - 1)
 
-        // asset.balance.amount += token.amount // update balance of the asset
+        asset.balance.amount += token.amount // update balance of the asset
 
         return {
             amount: issueAmount,
             denom: this.pool.supply.denom
         }
     }
 
-    // P_issued = P_supply * (1 + At/Bt)
+    // P_issued = P_supply * Wt * Dt/Bt
     function depositMultiAsset(tokens: Coin[]): Coin[] {
-        const lpTokens = [];
-        for (const token in tokens) {
-            const asset = this.pool.findAssetByDenom(token.denom)
-            const amount = token.amount
-            const supply = this.pool.supply.amount
-            const weight = asset.weight / 100
-            const issueAmount = supply * (1+amount/asset.balance)
-            // asset.balance.amount += token.amount // update balance of the asset
-            lpTokens.push({
-                amount: issueAmount,
-                denom: this.pool.supply.denom
-            });
+    const outTokens: Coin[] = [];
+    for (const token of tokens) {
+      const asset = imm.Pool.FindAssetByDenom(token.Denom);
+      if (!asset) {
+        throw new Error("Asset not found");
+      }
+
+      let issueAmount: Int;
+
+      if (imm.Pool.Status === PoolStatus_INITIALIZED) {
+        let totalAssetAmount = new Int(0);
+        for (const asset of imm.Pool.Assets) {
+          totalAssetAmount = totalAssetAmount.Add(asset.Balance.Amount);
         }
-        return lpTokens
+        issueAmount = totalAssetAmount.Mul(new Int(asset.Weight)).Quo(new Int(100));
+      } else {
+        const decToken = new DecCoinFromCoin(token);
+        const decAsset = new DecCoinFromCoin(asset.Balance);
+        const decSupply = new DecCoinFromCoin(imm.Pool.Supply);
+
+        const ratio = decToken.Amount.Quo(decAsset.Amount).Mul(new Dec(Multiplier));
+        issueAmount = decSupply.Amount.Mul(new Dec(asset.Weight)).Mul(ratio).Quo(new Dec(100)).Quo(new Dec(Multiplier)).RoundInt();
```

Comment:
It should use standard typescript code.

so, would suggest to use `+ - * / `

Reply:
We don't need `DecCoinFromCoin`, since it there's no definition. 
`Coin` is enough.